R/functions_util.R

```diff
@@ -654,28 +887,38 @@ check_installed_packages = function() {
 #' @param mirror Ensembl mirror.
 #' @param version Ensembl version.
 #' @param attributes The attributes to download.
+#' @param file_annot File with existing annotation to use instead.
 #' @return Returns a list with error messages.
-check_ensembl = function(biomart, dataset, mirror, version, attributes) {
+check_ensembl = function(biomart, dataset, mirror, version, attributes, file_annot=NULL) {
   error_messages = c()
-
-  # See if mart is available
-  annot_mart = suppressWarnings(GetBiomaRt(biomart, dataset, mirror, version))
-  if (is.null(annot_mart)) {
-    if (is.null(mirror)) {
-      return(paste0("Cannot download Ensembl annotation for dataset '",dataset,"', version '",version ," using biomaRt'!"))
-    } else {
-      return(paste0("Cannot download Ensembl annotation for dataset '",dataset,"', version '",version ,"' at mirror '",mirror,"' using biomaRt!"))
+  
+  if (is.null(file_annot) || !file.exists(file_annot)) {
```

Comment:
We need to check for Ensembl if either the annotation file or cell cycle file is not there.

Reply:
Done

Reply:
Done and the file name is now part of the param object (but cannot be set).